google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -73,7 +73,7 @@
   private static final String MANIFEST_ARTIFACT_ID_KEY = "artifactId";
   private static final String MANIFEST_VERSION_KEY = "Implementation-Version";
   private static final String ARTIFACT_ID = "google-cloud-core";
-  private static final String LIBRARY_NAME = "gcloud-java";
```

Comment:
Yes, Datastore is discovery-based in Java. Your double-negation didn't lead to a clear interpretation: are you stating that we need to implement `x-goog-api-client` for Datastore? Datastore is currently sending UA in Java.